missions/Controllers/Demo2/ActivityPredictionViewController.swift

```diff
@@ -129,3 +147,144 @@ extension ActivityPredictionViewController {
 //        })
     }
 }
+
+// Implemented by Elina
+
+@available(iOS 13, *)
+extension ActivityPredictionViewController {
+    
+    func setUpButtons() {
+        
+        setUpBackwardButton()
+        setUpForwardButton()
+        
+    }
+    
+    @IBAction func playButtonTapped(_ sender: UIButton) {
+        
+        if AudioPlayback.shared.audioPlayer.currentTime == AudioPlayback.shared.audioPlayer.duration {
+            AudioPlayback.shared.audioPlayer.currentTime = 0.00
+        } else {
+            self.playbackState = self.playbackState.toggle
+            self.playButton.setImage(self.playbackState.image, for: .normal)
+            updateRunLoop()
+        }
+        
+        AudioPlayback.shared.play(pan: stanceSlider.value)
+      
+      }
+    
+    @IBAction func onStanceChanged(_ sender: UISlider) {
+        
+        self.currentStance = Stance.stanceForValue(sender.value)
+        self.stanceLabel.text = self.currentStance.stanceLabelString
+        AudioPlayback.shared.setPanValue(sender.value)
+
+    }
+    
+    func updateRunLoop() {
+        
+        if updater == nil {
+            updater = CADisplayLink(target: self, selector: #selector(trackAudio(_:)))
+            updater.preferredFramesPerSecond = 10
+            updater.add(to: RunLoop.current, forMode: RunLoop.Mode.common)
+        }
+        
```

Comment:
Optional: same as the above comment. You don't need a newline between the closing brace of the method and the last line of executable code.